wallet/chainntfns.go

```diff
@@ -442,6 +442,7 @@ func (w *Wallet) birthdaySanityCheck() (*waddrmgr.BlockStamp, error) {
 		}
 
 		candidate.Hash = *hash
+		candidate.Height = int32(newCandidateHeight)
```

Comment:
can we add a test for this case?

Reply:
Agreed, we've caught two issues in this method that were lingering due to a lack of tests in this method. We can add a set of _minimal_ interfaces, so interfaces that contain _only_ the set of methods we need in order to minimize the foot print here.

Reply:
Tests added!